### PR TITLE
Beta/introduce near media query

### PR DESCRIPTION
**Pull request by tsmith023**

### CODE REVIEW

File: weaviate/collections/classes/grpc.py [hunk 1]
```diff
@@ -222,3 +223,14 @@ def __hash__(self) -> int:  # for set
 
 _PROPERTY = Union[PROPERTY, REFERENCE]
 _PROPERTIES = Union[PROPERTIES, REFERENCES]
+
+
+class NearMediaType(str, Enum):
+    """The different types of media that can be used in a `near_media` query to leverage the `multi2vec-bind` module."""
```
Comment (dirkkul) [inline weaviate/collections/classes/grpc.py:229]: I think there are more modules that take media (like Clip)

File: weaviate/collections/grpc/query.py [hunk 2]
```diff
@@ -334,71 +338,10 @@ def near_text(
...
-        self.__merge_return_references(return_references)
-
-        return self.__call()
-
-    def near_audio(
+    def near_media(
         self,
-        audio: str,
+        media: str,
+        type_: Literal["audio", "depth", "image", "imu", "thermal", "video"],
```
[the remark this answers is not in the document]
Comment (dirkkul) [reply weaviate/collections/grpc/query.py:344]: Also use the enum here?

**Review by dirkkul — CHANGES_REQUESTED** [review]
You didn't add the GRPC calls for the new near_XXX calls.

Looking at it again, should we also replace near_XXX with near_media + one extra type field? Not totally sure about it, as we could also have more fields for certain types in the future

**Comment by tsmith023** [comment]
>You didn't add the GRPC calls for the new near_XXX calls.

Ah, I thought I was being premature with the un-drafting but couldn't remember what I'd forgotten 😅 

> Looking at it again, should we also replace near_XXX with near_media + one extra type field? Not totally sure about it, as we could also have more fields for certain types in the future

I think it's good to have the gRPC messages as explicit as possible with implicit behaviour maintained within the client codebases themselves. Will help with flexibility in case of client specific nuances

**Review by dirkkul — APPROVED** [review]

**Pull request merged**